libs/wingc/src/type_check.rs

```diff
@@ -432,22 +432,14 @@ impl Subtype for Type {
 				}
 				false
 			}
-			(Self::Resource(l0), Self::Resource(_)) => {
-				// If we extend from `other` then I'm a subtype of it (inheritance)
-				if let Some(parent) = l0.parent.as_ref() {
-					let parent_type: &Type = parent;
-					return parent_type.is_subtype_of(other);
-				}
-				false
-			}
 			(Self::Interface(l0), Self::Interface(_)) => {
 				// If we extend from `other` then I'm a subtype of it (inheritance)
 				l0.extends.iter().any(|parent| {
 					let parent_type: &Type = parent;
 					parent_type.is_subtype_of(other)
 				})
 			}
-			(Self::Resource(res), Self::Interface(iface)) => {
+			(Self::Class(res), Self::Interface(iface)) => {
```

Comment:
Previously, the type checker had two separate types (Class and Resource) and now it only has one type (Class). Since this subtyping rule was only expected to work with the Resource type, we need to update the logic so that the subtype relationship does not hold if the class's phase is inflight.

For example, we should check that this is a compiler error:

```js
bring cloud;

inflight class A {
  handle(msg: str): str {
    return msg;
  }
}

// A does not automatically implement cloud.IFunctionHandler
// if A was a preflight class (and `handle` was an inflight method) then this would be ok
new cloud.Function(new A());
```

I think we just need to add something like:

```rust
if res.phase != Phase::Preflight {
	return false;
}
```

Reply:
nice catch

Reply:
So this currently fails with:

```
   |                    ^^^^^^^ Cannot create inflight class "A" in preflight phase
```

Which is a reasonable error, no?

But I guess I can add this if statement for completness (although impossible to test IMHO, no)?

Reply:
OK, on second thought, I think we don't have to limit this to preflight classes. I don't see a good reason for this structural subtyping not to work for inflight classes as well.
I generally think interface subtyping should be structural, but that's for a future discussion. In the meantime, I've added a test to verify this:

```js
bring cloud;

interface IGoo {
  inflight handle(): num;
}

inflight class NotGoo {
  handle(): num {
    return 123;
  }
}

test "structure interface types for 'handle'" {
  inflight class YesGoo impl IGoo {
    handle(): num {
      return 456;
    }
  }

  // this works
  let y: IGoo = new YesGoo();
  assert(y.handle() == 456);

  // this works too because if the interface has a `handle` method we use structural typing
  let x: IGoo = new NotGoo();
  assert(x.handle() == 123);
}

```

I generally prefer to reduce to a minimum the amount of special casing (preflight/inflight handle/no-handle). I think interfaces should be structurally typed acroos the system.